spec/unit/crypto/algorithms/megolm.spec.js

```diff
@@ -119,29 +119,32 @@ describe("MegolmDecryption", function() {
                 },
             };
 
-            expect(megolmDecryption.hasKeysForKeyRequest(keyRequest))
-                .toBe(true);
-
-            // set up some pre-conditions for the share call
-            const deviceInfo = {};
-            mockCrypto.getStoredDevice.andReturn(deviceInfo);
-
-            const awaitEnsureSessions = new Promise((res, rej) => {
-                mockOlmLib.ensureOlmSessionsForDevices.andCall(() => {
-                    res();
-                    return Promise.resolve({'@alice:foo': {'alidevice': {
-                        sessionId: 'alisession',
-                    }}});
+            return megolmDecryption.hasKeysForKeyRequest(
+                keyRequest,
+            ).then((hasKeys) => {
+                expect(hasKeys).toBe(true);
+
+                // set up some pre-conditions for the share call
+                const deviceInfo = {};
+                mockCrypto.getStoredDevice.andReturn(deviceInfo);
+
+                const awaitEnsureSessions = new Promise((res, rej) => {
```

Comment:
You never return this from the promise chain which is `return`ed to the test framework, so this test isn't waiting for all the stuff inside here to run.